src/perl/Permabit/KernelModule.pm

```diff
@@ -78,8 +79,16 @@ sub loadFromFiles {
   my ($self) = assertNumArgs(1, @_);
   my $modName = $self->{modName};
   my $modVer = $self->{modVersion};
-
-  my $loaded = $self->SUPER::loadFromFiles();
+  my $loaded = 0;
+
+  # With useUpstream test, we don't need to load upstream
+  # module(dm-vdo), since it is in the kernel already.
+  if ($self->{useUpstream} &&
+      $modName eq $self->{upstreamModName}) {
+    $loaded = 1;
```

Comment:
In this case, you could just "return 1;" here. I think it simplifies the logic a little.  Also, I don't think you need to check the modName. We should only be setting useUpsream for the dm-vdo module.